opensearch-operator/pkg/builders/cluster.go

```diff
@@ -176,8 +176,8 @@ func NewSTSForNodePool(
 		jvm = node.Jvm
 	}
 
-	// If node role `search` defined add required experimental flag
-	if helpers.ContainsString(selectedRoles, "search") {
+	// If node role `search` defined add required experimental flag if version less than 2.7
+	if helpers.ContainsString(selectedRoles, "search") && cr.Spec.General.Version < "2.7" {
```

Comment:
Please do not use a simple string comparison here as that does not always work correctly. See `ResolveClusterManagerRole` in `pkg/helpers/helpers.go` on how to do a proper version comparison.

Reply:
Hi, let me know if the way I have done it now is looking ok

Reply:
Looks good